src/lu_decomposition.c

```diff
@@ -0,0 +1,49 @@
+#include "lu_decomposition.h"
+#include "matrix.h"
+#include <stdio.h>
+#include <string.h>
+
+/**
+ * @brief Perform the LU decomposition algorithm on the matrix initialMatrix
+ *
+ * @param initialMatrix The matrix to decompose
+ * @param lMatrix Output matrix L
+ * @param uMatrix Output matrix U
+ * @param size Size of the matrix
+ */
+void LUDecomposition(const lu_real* initialMatrix, lu_real* lMatrix,
+                     lu_real* uMatrix, const int size) {
```

Comment:
you made me thought of this (i'm not sure we have been doing it so far), but aren't output parameters usually declared as the last parameters ?

Reply:
This is not something that we usually do.
I did implement this logic in the lanczos algorithm, but most algorithms are not using the last parameter as a return parameter. (Take a look, for example, at gradient descent or genetic). 

I could change the order of my parameters for this method if you think this is something that we should do.